src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -5824,7 +5824,7 @@ private ScanTypeArgumentListKind ScanTypeArgumentList(NameOptions options)
                         // Note: we check if we got 'MustBeType' which triggers for predefined types,
                         // (int, string, etc.), or array types (Goo[], A<T>[][] etc.), or pointer types
                         // of things that must be types (int*, void**, etc.).
-                        isDefinitelyTypeArgumentList = DetermineIfDefinitelyTypeArgumentList(isDefinitelyTypeArgumentList);
+                        isDefinitelyTypeArgumentList = isDefinitelyTypeArgumentList || this.CurrentToken.Kind is SyntaxKind.CommaToken or SyntaxKind.GreaterThanToken;
```

Comment:
i inlined this method as i found it easier to reason about just looking at teh check directly in code.

Reply:
note: tehre are further changes i'd like to make to this method for clarity/correctness/invariants.  but i'm trying to keep the PR small for the purposes of fixing this important issue.